src/Controls/src/Build.Tasks/XmlTypeExtensions.cs

```diff
@@ -85,7 +85,10 @@ public static bool TryGetTypeReference(this XmlType xmlType, XamlCache cache, Mo
 			TypeReference type = xmlType.GetTypeReference(xmlnsDefinitions, module.Assembly.Name.Name, (typeInfo) =>
 			{
 				string typeName = typeInfo.typeName.Replace('+', '/'); //Nested types
-				return module.GetTypeDefinition(cache, (typeInfo.assemblyName, typeInfo.clrNamespace, typeName));
+				var type = module.GetTypeDefinition(cache, (typeInfo.assemblyName, typeInfo.clrNamespace, typeName));
+				if (type is not null && type.IsPublicOrVisibleInternal(module))
+					return type;
+				return null;
```

Comment:
This is the real fix for XamlG. Make sure the type is visible to the assembly referencing it.